fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/lao/witness/WitnessMessageListViewAdapter.java

```diff
@@ -111,8 +112,20 @@ public View getView(int position, View convertView, ViewGroup parent) {
 
     if (isWitness) {
       Context context = parent.getContext();
-      View.OnClickListener listener = setUpSignButtonClickListener(context, witnessMessage);
-      binding.signMessageButton.setOnClickListener(listener);
+      Set<PublicKey> witnessSignatures = witnessMessage.getWitnesses();
+
+      if (witnessSignatures.contains(laoViewModel.getPublicKey())) {
+        // The user already signed the message
+        binding.signMessageButton.setEnabled(false);
+        binding.signMessageButton.setText("Signed");
+      } else {
+        binding.signMessageButton.setEnabled(true);
+        binding.signMessageButton.setText("Sign");
+        View.OnClickListener listener =
```

Comment:
Last thing that I forgot, could you use string resources here? As a general principle each displayed string should be an Android resource